### DISCUSSION TITLE
Need help with nested fields of type "object". TinaCMS throws an error on pages with blank front matter

### DISCUSSION THREAD

**Discussion by jobee** [tinacms/tinacms#4030] [Help]
I try to make Hugo`s front matter for menus editable in TinaCMS.

The front matter structure expects something like this:

```yaml
menu:
  main:
    title: "Page title in menu"
    parent: "Products"
    pre: "<i class="fa-solid fa-code"></i>"
    weight: 20
```
See https://gohugo.io/content-management/menus/#example-front-matter

My current fields config (nested fields of type "object") works, if the page already includes the outer `menu` key in front matter.
On new pages Tina throws an error, e.g. when I try to edit the fields within "main".

```
TinaCMS Render Error
Tina caught an error while updating the page:

Cannot read properties of undefined (reading 'main')
```

How can this be avoided?

Here's my field config:

```typescript
export function menuFields() {
  return [
    {
      type: "object",
      name: "menu",
      label: "Navigation: Include this page in menu(s)",
      fields: [
        {
          type: "object",
          name: "main",
          label: "Main Menu",
          fields: [...menuProperties()],
        },
        {
          type: "object",
          name: "meta",
          label: "Meta Menu",
          fields: [...menuProperties()],
        },
        {
          type: "object",
          name: "footer",
          label: "Footer Menu",
          fields: [...menuProperties()],
        },
      ],
    },
  ] as TinaField[];
}

function menuProperties() {
  return [
    {
      type: "string",
      name: "identifier",
      label: "Identifier",
    }, 
    {
      type: "string",
      name: "name",
      label: "Name",
    },
    {
      type: "string",
      name: "parent",
      label: "Parent",
    },
    {
      type: "string",
      name: "post",
      label: "Post",
    },
    {
      type: "string",
      name: "pre",
      label: "Pre",
    },
    {
      type: "string",
      name: "title",
      label: "Title",
    },
    {
      type: "number",
      name: "weight",
      label: "Weight",
    },
  ] as TinaField[];
}
```

**Comment by jeffsee55** [comment]
You might not be receiving a helpful error message but at least one thing that doesn't look quite right is the `parent` field. It's of type `string` when it should be an `object` with a `class` field in it.

**Reply by jobee** [reply]
@jeffsee55 thanks for your quick reply. You're right that `params` probably needs an other field type. But that's not the issue I'm facing. Actually my site is not using `params` at all, I've updated my example. Sorry for the confusion.

It's mainly about editing nested objects in Tina, when the key of the parent object is not yet present in front matter.

### Let me simply my example here:

1.) The markdown file of a new page looks like:

```md
---
title: New page
---

...
```

2.) Now I try to set menu format in front matter using the following Tina field config:

```typescript
export function menuFields() {
  return [
    {
      type: "object",
      name: "menu",
      label: "Navigation: Include this page in menu(s)",
      fields: [
        {
          type: "object",
          name: "main",
          label: "Main Menu",
          fields: [
            {
              type: "string",
              name: "title",
              label: "Title",
            },
            {
              type: "number",
              name: "weight",
              label: "Weight",
            },
          ],
        },
      ],
    },
  ] as TinaField[];
}
```

3.) When I try to edit the `main` object, Tina throws an error:

```
TinaCMS Render Error
Tina caught an error while updating the page:

Cannot read properties of undefined (reading 'main')
```

4.) If I set the parent object key (in this example `menu`) manually, like:

```md
---
title: New page
menu: false
---

...
```

5.) Tina would let me edit the nested object (is this example: `main`) and save to the file, which results in:

```md
---
title: New page
menu:
  main:
    title: Title in menu
    weight: 10
---

...
```
